### PR TITLE
Fix #873, Check previous version for Notes unless they are First Public Note

### LINKED ISSUE #873: Check previous version for Notes unless they are First Public WG Note
At the moment, we don't expect Notes to have a previous version. However, this may happen and if it was previously published, we need to make sure the previous version is specified and is the right one.

### PR DESCRIPTION
**Pull request by jennyliang220**

### CODE REVIEW

File: lib/rules/metadata/profile.js [hunk 1]
```diff
@@ -68,8 +72,46 @@ exports.check = function(sr, done) {
...
+            var $linkLatest = (dts.Latest) ? dts.Latest.dd.find("a").first().attr('href') : '';
+            shortname = $linkLatest.match(/^https:\/\/www.w3.org\/TR\/(.+)\/$/)[1];
+            var req = sua.get('https://api.w3.org/specifications/' + shortname)
+                .set("User-Agent", "W3C-Pubrules/" + sr.version);
+            req.query({ apikey: process.env.W3C_API_KEY });
+            req.end(function (err, res) {
+                if (err || !res.ok) {
+                    if (res && res.status == 404) {
+                        specExists = false;
+                    }
```
Comment (deniak) [inline lib/rules/metadata/profile.js:96]: There could actually be other kind of errors other than 404 (e.g. network error, 500, etc). I guess in this case, we can simply assume the spec doesn't exist.

```suggestion
                    specExists = false;
```

File: lib/rules/metadata/profile.js [hunk 2]
```diff
@@ -68,8 +72,46 @@ exports.check = function(sr, done) {
...
+        return new PowerPromise(function (resolve) {
+            var $dl = sr.$("body div.head dl").first()
+            ,   dts = sr.extractHeaders($dl)
+            ,   $linkPrev = (dts.Previous) ? dts.Previous.dd.find("a").first() : ''
+            ,   shortname
+            ,   specExists
+            ;
+
+            var $linkLatest = (dts.Latest) ? dts.Latest.dd.find("a").first().attr('href') : '';
+            shortname = $linkLatest.match(/^https:\/\/www.w3.org\/TR\/(.+)\/$/)[1];
```
Comment (deniak) [inline lib/rules/metadata/profile.js:88]: Check if the index 1 actually exists. There might be a parsing issue

File: lib/rules/metadata/profile.js [hunk 3]
```diff
@@ -68,8 +72,46 @@ exports.check = function(sr, done) {
...
+                } else {
+                    specExists = true;
+                }
+                resolve(specExists || $linkPrev);
+            });
+        });
+    };
+
+    if (id) {
+        if (/-NOTE/.test(id)) {
```
Comment (deniak) [inline lib/rules/metadata/profile.js:106]:
```suggestion
        if (/\-NOTE/.test(id)) {
```

Comment (jennyliang220) [reply lib/rules/metadata/profile.js:105]: After adding escape `\`, build failed: https://travis-ci.com/github/w3c/specberus/builds/166576468

```
/home/travis/build/w3c/specberus/lib/rules/metadata/profile.js
  105:14  error  Unnecessary escape character: \-  no-useless-escape
```

Comment (deniak) [reply lib/rules/metadata/profile.js:105]: Damn `eslint`! Ok let's skip that "useless" escape :)

**Pull request merged**